libexecutive/Executive.cpp

```diff
@@ -382,6 +367,62 @@ bool Executive::executeCreate(Address const& _sender, u256 const& _endowment, u2
     return !m_ext;
 }
 
+void Executive::grantContractStatusManager(TableFactory::Ptr memoryTableFactory,
+    Address const& newAddress, Address const& sender, Address const& origin)
+{
+    LOG(TRACE) << LOG_DESC("grantContractStatusManager") << LOG_KV("contract", newAddress)
+               << LOG_KV("sender account", sender) << LOG_KV("origin account", origin);
+
+    std::string tableName = precompiled::getContractTableName(newAddress);
+    auto table = memoryTableFactory->openTable(tableName);
+
+    if (!table)
+    {
+        LOG(ERROR) << LOG_DESC("grantContractStatusManager get newAddress table error!");
+        return;
+    }
+
+    // grant origin authorization
+    auto entry = table->newEntry();
+    entry->setField("key", "authority");
```

Comment:
Since it is called multiple times here, it is recommended to encapsulate it into a function